src/SearchWrapper.cpp

```diff
@@ -0,0 +1,163 @@
+
+using namespace std;
+
+Rcpp::List convertToDataFrame(Rcpp::List list) {
+  Rcpp::List dataFrame = clone(list);
+  Rcpp::GenericVector aRow = dataFrame(0);
+  Rcpp::StringVector rowNames(aRow.length());
+  
+  for (int i = 0; i < aRow.length(); ++i) {
+    char name[5];
+    sprintf(&(name[0]), "%d", i+1);
+    rowNames(i) = name;
+  }
+  
+  dataFrame.attr("row.names") = rowNames;
+  Rcpp::StringVector colNames(dataFrame.length());
+  
+  int ascii=97;
+  for (int j = 0; j < dataFrame.length(); ++j) {
+    char name[6];
+    //sprintf(&(name[0]), "%d", j+1);
+    sprintf(&(name[0]), "%c", char(ascii++));
+    colNames(j) = name;
+  }
+  
+  dataFrame.attr("names") = colNames;
+  dataFrame.attr("class") = "data.frame";
+  return dataFrame;
+}
+
+int columnNo( const string & data){
+  int col=0;
+  std::istringstream stream(data);
+  std::string line;
+  while(std::getline(stream, line)) {
+    col++;
+  }
+  return col;
+}
+
+string rowVal(const string & data,int row){
+  int r=0;
+  string val;
+  std::istringstream stream(data);
+  std::string line;
+  while(std::getline(stream, line)) {
+    if(r==row){
+      val=line;
+      break; 
+    }
+    r++;
+  }
+  if(val=="")
+    return "NA";
+  else
+    return val;
+}
+
+Xapian::Query parseQuery(Rcpp::List & query){
+  Xapian::Query returnQuery;
+  if(query.containsElementNamed("queryString")){
+    
+    Xapian::QueryParser queryparser;
+    
+    if(query.containsElementNamed("stemmer")){
+      std::string lang=Rcpp::as<std::string>(query["stemmer"]);
+      queryparser.set_stemmer(Xapian::Stem(lang));
+    }
+    
+    if(query.containsElementNamed("strategy")){
+      //maybe an if else should be used here
+    }
+    else
+      queryparser.set_stemming_strategy(queryparser.STEM_SOME);
+    
+    if(query.containsElementNamed("prefix.fields")){
+      Rcpp::List prefixFields=query["prefix.fields"];
+      for( Rcpp::List::iterator it = prefixFields.begin(); it != prefixFields.end(); ++it ) {
+        Rcpp::List list=it.operator*();
+        std::string prefix=Rcpp::as<std::string>(list["prefix"]);
+        std::string fieldName=Rcpp::as<std::string>(list["name"]);
+        queryparser.add_prefix(fieldName, prefix);
+      }
+    }
+    
+    std::string aQuery=Rcpp::as<std::string>(query["queryString"]);
+    returnQuery = queryparser.parse_query(aQuery);
+  }
+  
+  return returnQuery;
+}
+
+//' Search wrapper
+//'
+//' @param dbpath	path to a Xapian database
+//' @param queryList    a list of data to create a Xapian::Query 
+//' 
+//' @examples
+//' \dontrun{
+//' db<- c("path/to/database")
+//' query<-list(queryString="watch",
+//'             prefix.fields=list(list(prefix="S",name="title"),
+//'                           list(prefix="XD",name="description")),
+//'             stemmer="en")
+//' searchWrapper(db,query)
+//' }
+//' 
+//' @return 
+// [[Rcpp::export]]
+Rcpp::List searchWrapper(Rcpp::CharacterVector & dbpath,Rcpp::List & queryList){
+  Rcpp::List aQuery=queryList;
```

Comment:
This comes in as a reference, and is passed to `parseQuery` as a reference, so do you need to copy it into a normal variable in the middle?